velox/type/Type.h

```diff
@@ -2009,6 +2009,34 @@ bool registerCustomType(
     const std::string& name,
     std::unique_ptr<const CustomTypeFactories> factories);
 
+// See registerOpaqueType() for documentation on type index and opaque type
+// alias.
+std::unordered_map<std::string, std::type_index>& getTypeIndexByOpaqueAlias();
+
+// Reverse of getTypeIndexByOpaqueAlias() when we need to look up the opaque
+// alias by its type index.
+std::unordered_map<std::type_index, std::string>& getOpaqueAliasByTypeIndex();
+
+std::type_index getTypeIdForOpaqueTypeAlias(const std::string& name);
+
+std::string getOpaqueAliasForTypeId(std::type_index typeIndex);
+
+// OpaqueType represents a type that is transparent to the Velox type system.
```

Comment:
nit: I think the point is that opaque types are NOT transparent to the type system. :) (they are "opaque" to the type system) :)